software/control/widgets.py

```diff
@@ -2679,6 +2679,29 @@ def toggle_z_range_controls(self, state):
         self.updateGeometry()
         self.update()
 
+    def init_z(self, z_pos_mm=None):
+        if z_pos_mm is None:
+            z_pos_mm = self.navigationController.z_pos_mm
+
+        # block entry update signals
+        self.entry_minZ.blockSignals(True)
+        self.entry_maxZ.blockSignals(True)
+
+        try: # disconnect any previous connections updating the z min and max from the current z pos
+            self.navigationController.zPos.disconnect(self.update_z_min)
+            self.navigationController.zPos.disconnect(self.update_z_max)
```

Comment:
@sohamazing when are the signals reconnected?

Reply:
Initially the z-min & z-max would update as we move the z-pos on the microcontroller. I then added the init_z method so that when we finish the start up we then set the initial z-range values. This was because there was a difference between the z pos when we create the widget and the z pos when we finish loading the gui. with the current implemntation where we do not update the z-range ever with the microcontroller directly we can remove these signal disconnects and try/catch blocks